service.go

```diff
@@ -296,24 +296,44 @@ func (svc *Service) ProcessPayment(ctx context.Context, payment *lnrpc.Payment)
 	//if the payment was in the database as final then we already published it
 	//and we only publish completed payments
 	if notInflight && !alreadyPublished {
-		logrus.Infof("Publishing payment status %v hash %s", payment.Status, payment.PaymentHash)
 		err := svc.PublishPayload(ctx, payment, LNDPaymentExchange, routingKey)
 		if err != nil {
+			logrus.WithFields(
+				logrus.Fields{
+					"payload_type": "payment",
+					"status":       fmt.Sprintf("%s", payment.Status),
+					"payment_hash": payment.PaymentHash,
+				}).WithError(err).Info("error publishing payment")
 			tx.Rollback()
 			return err
 		}
+		logrus.WithFields(
+			logrus.Fields{
+				"payload_type": "payment",
+				"status":       fmt.Sprintf("%s", payment.Status),
+				"payment_hash": payment.PaymentHash,
+			}).Info("published payment")
 	}
 
 	return tx.Commit().Error
 }
 
 func (svc *Service) ProcessInvoice(ctx context.Context, invoice *lnrpc.Invoice) error {
 	if invoice.State == lnrpc.Invoice_SETTLED {
-		logrus.Infof("Publishing invoice with hash %s", hex.EncodeToString(invoice.RHash))
 		err := svc.PublishPayload(ctx, invoice, LNDInvoiceExchange, LNDInvoiceRoutingKey)
 		if err != nil {
+			logrus.WithFields(
+				logrus.Fields{
+					"payload_type": "invoice",
+					"payment_hash": hex.EncodeToString(invoice.RHash),
+				}).WithError(err).Info("error publishing invoice")
```

Comment:
`.Error`?


and is it maybe useful to add more metdata/categorization from the invoice?  (amount, destination, ?)